app/pkg/dbx/migrate.go

```diff
@@ -140,3 +144,27 @@ func getLastMigration() (int, error) {
 
 	return int(lastVersion.Int64), nil
 }
+
+func getPendingMigrations(versions []int) ([]int, error) {
+	versionStr := ""
+	for _, version := range versions {
+		versionStr = versionStr + "," + strconv.Itoa(version)
+	}
+	versionStr = versionStr[1:]
+
+	rows, err := conn.Query("SELECT version FROM migrations_history WHERE version IN (" + versionStr + ")")
+	if err != nil {
+		return nil, err
+	}
+
+	for rows.Next() {
+		var version int
+		_ = rows.Scan(&version)
+		i := slices.Index(versions, version)
+		if i != -1 {
+			versions = slices.Delete(versions, i, i+1)
+		}
+	}
```

Comment:
In go it's generally bad practice to directly modify the input params, unless this is intended. I would copy versions into a new slice at the start of the function, and work with that new slice, rather than modify slices.

Reply:
Makes sense, I change it !